model/vfs/vfs.go

```diff
@@ -870,3 +879,56 @@ func OptionsAllowCreationInTrash(opts []CreateOptions) bool {
 	}
 	return false
 }
+
+func CheckAvailableDiskSpace(fs VFS, doc *FileDoc) (newsize, maxsize, capsize int64, err error) {
+	diskQuota := fs.DiskQuota()
+
+	maxsize = fs.MaxFileSize()
+	newsize = doc.ByteSize
+	if diskQuota > 0 {
+		diskUsage, err := fs.DiskUsage()
+		if err != nil {
+			return 0, 0, 0, err
+		}
+		maxsize = diskQuota - diskUsage
```

Comment:
I think we have lost a case in extracting `CheckAvailableDiskSpace`. If the instance has a 50GB disk quota, with 1GB taken, `maxsize` should be 5GB when using VFS Swift, not 49GB (`diskQuota - diskUsage`).

Reply:
I'll take a look at this.